File: maxhanna.client/src/services/datacontracts/crypto/mining-rig.ts
import { MiningRigDevice } from "./mining-rig-device";

export class MiningRig {
  rigId: string | undefined;
  rigName: string | undefined; 
  minerStatus: string | undefined;
  unpaidAmount: number | undefined;
  speedRejected: number | undefined;
  localProfitability: number | undefined;
  actualProfitability: number | undefined;
  devices: Array<MiningRigDevice> | undefined;
}

File: maxhanna.Server/Controllers/DataContracts/Crypto/MiningRig.cs
namespace maxhanna.Server.Controllers.DataContracts.Crypto
{
	public class MiningRig
	{
		public string? rigId { get; set; }
		public string? rigName { get; set; }
		public string? minerStatus { get; set; }
		public float unpaidAmount { get; set; }
		public float speedRejected { get; set; }
		public float localProfitability { get; set; }
		public float actualProfitability { get; set; }
		public List<MiningRigDevice>? devices { get; set; }
	}
}
